joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller.hpp

```diff
@@ -231,13 +240,29 @@ class JointTrajectoryController : public controller_interface::ControllerInterfa
     const JointTrajectoryPoint & current_state,
     const JointTrajectoryPoint & state_error);
 
+  void read_state_from_hardware(JointTrajectoryPoint & state);
+
+  bool read_state_from_command_interfaces(JointTrajectoryPoint & state);
+
 private:
   bool contains_interface_type(
     const std::vector<std::string> & interface_type_list, const std::string & interface_type)
   {
     return std::find(interface_type_list.begin(), interface_type_list.end(), interface_type) !=
            interface_type_list.end();
   }
+
+  void resize_joint_trajectory_point(
+    trajectory_msgs::msg::JointTrajectoryPoint & point, size_t size)
+  {
+    point.positions.resize(size);
+    if (has_velocity_state_interface_) {
+      point.velocities.resize(size);
+    }
+    if (has_acceleration_state_interface_) {
+      point.accelerations.resize(size);
+    }
+  }
```

Comment:
nitpick, is there any reason why this is on the header?

Reply:
Not really. I simply thought it's a basic utility function, so I put it there. Should I move this to cpp file? Or should I create a utility namespace outside of the class for this?

Reply:
Move to the cpp file for sure, otherwise if this gets included twice it will produce a linker double definition error.